src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6815,6 +6815,12 @@ private bool IsPossibleLocalDeclarationStatement(bool allowAnyExpression)
                 return true;
             }
 
+            // deconstruction-assignment
+            if (tk == SyntaxKind.OpenParenToken)
```

Comment:
For the benefits of the group: Neal and I had some brainstorming on this. The parsing is definitely going to be trickier. More on this after the weekend.